internal/reporter.go

```diff
@@ -32,22 +32,38 @@ func (reporter reporter) Report(format string, pointLines string) (*http.Respons
 		return nil, formatError
 	}
 
-	// compress
+	buf, err := linesToBuffer(pointLines)
+	if err != nil {
+		return &http.Response{}, err
+	}
+
+	req, err := reporter.buildRequest(format, buf)
+	if err != nil {
+		return &http.Response{}, err
+	}
+
+	return reporter.execute(req)
+}
+
+func linesToBuffer(pointLines string) (bytes.Buffer, error) {
 	var buf bytes.Buffer
 	zw := gzip.NewWriter(&buf)
 	_, err := zw.Write([]byte(pointLines))
 	if err != nil {
 		zw.Close()
-		return nil, err
+		return bytes.Buffer{}, err
 	}
 	if err = zw.Close(); err != nil {
-		return nil, err
+		return bytes.Buffer{}, err
 	}
+	return buf, err
+}
 
+func (reporter reporter) buildRequest(format string, buf bytes.Buffer) (*http.Request, error) {
```

Comment:
Instead of passing in a bytes.Buffer, pass in a simple `[]byte` instead. This makes buildRequest() easier to use.